src/components/flyout/deviceReconfigureFlyout.js

```diff
@@ -32,14 +42,46 @@ class DeviceReconfigureFlyout extends React.Component {
     this.commonConfigValueChanged = this.commonConfigValueChanged.bind(this);
     this.onChangeInput = this.onChangeInput.bind(this);
     this.applyDeviceConfigureJobsData =this.applyDeviceConfigureJobsData.bind(this);
+    this.checkJobStatus = this.checkJobStatus.bind(this);
   }
 
-  componentWillMount() {
-    this.calcCommonConfiguration(this.props.devices);
+  componentDidMount() {
+    const { devices, propertyUpdateJobs } = this.props;
+    this.checkJobStatus(devices, propertyUpdateJobs);
+    this.calcCommonConfiguration(devices);
   }
 
   componentWillReceiveProps(nextProps) {
-    this.calcCommonConfiguration(nextProps.devices);
+    const { devices, propertyUpdateJobs } = nextProps;
+    if (!_.isEqual(propertyUpdateJobs, this.props.propertyUpdateJobs)) {
+      this.checkJobStatus(devices, propertyUpdateJobs);
+    }
+    this.calcCommonConfiguration(devices);
+  }
+
+  checkJobStatus (devices, propertyUpdateJobs) {
+    const jobs = getRelatedJobs(devices, propertyUpdateJobs);
+    const jobStream = Rx.Observable.from(jobs);
+    jobStream
+      .flatMap(({ jobId, deviceIds }) =>
+        Rx.Observable
+          .fromPromise(ApiService.getJobStatus(jobId))
+          // Get completed jobs
+          .filter(({ status }) => status === 3)
+          .map(job => ({...job, deviceIds}))
+          // Get device from iothubmanager
+          .flatMap(({ deviceIds }) => Rx.Observable
+            .from(deviceIds)
+            .flatMap(deviceId => Rx.Observable
+              .fromPromise(ApiService.getDeviceById(deviceId))
+            )
+            .reduce((acc, cur) => acc.concat([cur]), [])
+          )
+      )
+      .subscribe(
```

Comment:
I think this could be a bit cleaner. Is there any chance that device Ids could be duplicated in this stream?

```
Rx.Observable.from(jobs)
  .flatMap(({ jobId, deviceIds }) =>
    Rx.Observable
      .fromPromise(ApiService.getJobStatus(jobId))
      .filter(({ status }) => status === 3)
      .flatMap(_ => deviceIds)
  )
  .flatMap(deviceId => 
    Rx.Observable
      .fromPromise(ApiService.getDeviceById(deviceId))
  )
  .reduce((devices, device) => [...devices, device], [])
  .subscribe(
    devices => this.props.actions.updateDevices(devices),
    error => console.log('error', error)
  );
```